charts/argo-cd/templates/argocd-configs/argocd-cmd-params-cm.yaml

```diff
@@ -0,0 +1,106 @@
+{{- $redisHa := (index .Values "redis-ha") -}}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: argocd-cmd-params-cm
+  labels:
+    {{- include "argo-cd.labels" (dict "context" . "component" .Values.server.name "name" "cmd-params-cm") | nindent 4 }}
+data:
+  # Repo server address. (default "argocd-repo-server:8081")
+  repo.server: "{{ template "argo-cd.repoServer.fullname" . }}:{{ .Values.repoServer.service.port }}"
+  {{- if .Values.dex.enabled }}
+  # Dex server address (default "http://argocd-dex-server:5556")
+  dex.server: "http://{{ template "argo-cd.dex.fullname" . }}:{{ .Values.dex.servicePortHttp }}"
+  {{- end }}
+  {{- if or (and .Values.redis.enabled (not $redisHa.enabled)) (and $redisHa.enabled $redisHa.haproxy.enabled) }}
+  # Redis server hostname and port (e.g. argocd-redis:6379)
+  redis.server: "{{ template "argo-cd.redis.fullname" . }}:{{ .Values.redis.servicePort }}"
+  {{- else if and .Values.externalRedis.host .Values.externalRedis.port }}
+  redis.server: "{{ .Values.externalRedis.host }}:{{ .Values.externalRedis.port }}"
+  {{- end }}
+
+  ## Controller Properties
+  # Number of application status processors (default 20)
+  controller.status.processors: "{{ .Values.controller.args.statusProcessors }}"
```

Comment:
I think we should prepare a breaking change (major version) and refactor it. It looks kinda strange to do a random variable mapping here. Eg. `controller.status.processors = controller.args.statusProcessors`


I'd prefer a structure like:
```yaml
configs:
  cm: # or config: / configmap: / ...?
    # <keys that goes into argocd-cm>

  params:
    # <keys that goes into argocd-cmd-params-cm>
```
Maybe we can use a template function for that and merge the resulting yaml with our templated things like 

- `dex.server: "http://{{ template "argo-cd.dex.fullname" . }}:{{ .Values.dex.servicePortHttp }}"`
- `  redis.server: "{{ template "argo-cd.redis.fullname" . }}:{{ .Values.redis.servicePort }}"`
- etc

Reply:
I'm open about such change. If I understand correctly you would like to consolidate everything under `configs` section including stuff that is currently under `server`? And after that merge user config with internally set keys such as dex, redis, etc.?